auth/auth.go

```diff
@@ -0,0 +1,281 @@
+package auth
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-oidc"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/scopes"
+)
+
+// Config contains the configuration for the SAMS authentication handler.
+type Config struct {
+	// Issuer is the SAMS instance URL, e.g. "https://accounts.sourcegraph.com".
+	Issuer string
+	// ClientID is the SAMS client ID, e.g. "sams_cid_xxx".
+	ClientID string
+	// ClientSecret is the SAMS client secret, e.g. "sams_cs_xxx".
+	ClientSecret string
+	// RequestScopes is the list of requested scopes for access tokens that are
+	// issued to this client.
```

Comment:
Issued to the _users_ who log in using this SAMS client configuration?

Reply:
Admittedly grammar is so hard here... 😂 Users never gonna see those access tokens right? So technically those are indeed issues to the client. 🤔 Happy to accept anything a native speaker suggests here...